rust/libs/connlib/phoenix-channel/src/lib.rs

```diff
@@ -1114,4 +1190,104 @@ mod tests {
         .unwrap()
         .unwrap();
     }
+
+    #[tokio::test]
+    async fn http_429_triggers_retry() {
+        let port = http_status_server(429, "Too Many Requests").await;
+
+        let mut channel = make_test_channel(port);
+        channel.connect(PublicKeyParam([0u8; 32]));
+
+        let result = tokio::time::timeout(Duration::from_secs(5), async {
+            future::poll_fn(|cx| channel.poll(cx)).await
+        })
+        .await
+        .expect("should not timeout");
+
+        // 429 should trigger a Hiccup (retry) not an Error::Client
+        assert!(
+            matches!(result, Ok(Event::Hiccup { .. })),
+            "expected Event::Hiccup for 429, got {result:?}"
+        );
+    }
+
+    #[tokio::test]
+    async fn http_408_triggers_retry() {
+        let port = http_status_server(408, "Request Timeout").await;
+
+        let mut channel = make_test_channel(port);
+        channel.connect(PublicKeyParam([0u8; 32]));
+
+        let result = tokio::time::timeout(Duration::from_secs(5), async {
+            future::poll_fn(|cx| channel.poll(cx)).await
+        })
+        .await
+        .expect("should not timeout");
+
+        // 408 should trigger a Hiccup (retry) not an Error::Client
```

Comment:
I have not seen this before. When are we sending 408s?

Reply:
This is apparently commonly returned by load balancers if the client opens a connection and does not send any HTTP data in time.

From https://docs.cloud.google.com/load-balancing/docs/https/https-logging-monitoring#failure-messages -

> The Google Front End (GFE) idled out the client connection due to lack of progress while it was proxying either the request or the response.

I thought it's a good idea to add a handler for this in case another bug or specific network condition triggers this response.